core/src/services/gcs/core.rs

```diff
@@ -318,16 +316,22 @@ impl GcsCore {
 
         let mut req = Request::put(&url);
 
+        if let Some(user_metadata) = args.user_metadata() {
+            for (key, value) in user_metadata {
```

Comment:
Should we keep this? In the `gcs_insert_object_request` we are using the multipart body now... I don't know if this would work

Reply:
XML API is different from JSON API; we need to keep this.